src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/InfosetObjects/XamlNodes.cs

```diff
@@ -64,18 +64,9 @@ public XamlNode(XamlNodeType nodeType, object data)
 #if DEBUG
             switch(nodeType)
             {
+            case XamlNodeType.NamespaceDeclaration:
             case XamlNodeType.StartObject:
-                Debug.Assert(data is XamlType, "XamlNode ctor, StartObject data is not a XamlType");
-                break;
-
             case XamlNodeType.StartMember:
-                Debug.Assert(data is XamlMember, "XamlNode ctor, StartMember data is not a XamlMember");
```

Comment:
@hughbe Can we leave the ```Debug.Assert``` statement for now?
We do need them for local debugging. 
We will plan to remove them in future.